CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+cmake_minimum_required(VERSION 3.16)
+
+include(FetchContent)
+
+if (NOT CMAKE_TOOLCHAIN_FILE)
+    set(CMAKE_TOOLCHAIN_FILE
+        "${CMAKE_CURRENT_SOURCE_DIR}/build/emsdk/upstream/emscripten/cmake/Modules/Platform/\
+Emscripten.cmake"
+    )
+endif ()
+
+project(ClpIrStreamReader
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20
+# sources.
+if (NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR
+        "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif ()
+
+# Enable compile commands by default if the generator supports it.
+if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
```

Comment:
I believe this is to respect the setting if specified by the user / IDE. I removed the check and did not find any anomaly in CLion.

Reply:
Ah, then I think we should've used `NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS` since I think `NOT CMAKE_EXPORT_COMPILE_COMMANDS` will evaluate `CMAKE_EXPORT_COMPILE_COMMANDS` and if it's undefined/OFF, it will run the block (which is the same as not having the condition) and if it's on, then it won't run.